compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolResolver.java

```diff
@@ -1942,6 +1942,12 @@ public BSymbol getUnaryOpsForTypeSets(OperatorKind opKind, BType type) {
         if (!validNumericTypeExists) {
             return symTable.notFoundSymbol;
         }
+        if (opKind == OperatorKind.ADD) {
+            if (type.isNullable()) {
+                return createUnaryOperator(opKind, type, BUnionType.create(null, type, symTable.nilType));
```

Comment:
Why do we need to create a union type here? Can't we just return `createUnaryOperator(opKind, type, type)` if `opKind == OperatorKind.ADD`?

Reply:
Yeah, no need of creating it again. We can just return the type. Will change that